gcc/rust/resolve/rust-ast-resolve-expr.h

```diff
@@ -42,16 +42,39 @@ class ResolvePath : public ResolverBase
     resolver.resolve_path (expr);
   }
 
+  static void go (AST::SimplePath *expr, NodeId parent)
+  {
+    ResolvePath resolver (parent);
+    resolver.resolve_path (expr);
+  }
+
 private:
   ResolvePath (NodeId parent) : ResolverBase (parent) {}
 
   void resolve_path (AST::PathInExpression *expr);
 
   void resolve_path (AST::QualifiedPathInExpression *expr);
 
+  void resolve_path (AST::SimplePath *expr);
+
   void resolve_segments (CanonicalPath prefix, size_t offs,
 			 std::vector<AST::PathExprSegment> &segs,
 			 NodeId expr_node_id, Location expr_locus);
+
+  void
+  resolve_simple_path_segments (CanonicalPath prefix, size_t offs,
+				const std::vector<AST::SimplePathSegment> &segs,
+				NodeId expr_node_id, Location expr_locus);
+};
+
+class ResolveSimplePathSegmentToCanonicalPath
+{
+public:
+  static CanonicalPath resolve (const AST::SimplePathSegment &seg)
+  {
+    // FIXME: Since this is so simple, maybe it can simply be a tiny function?
+    return CanonicalPath::new_seg (seg.get_node_id (), seg.get_segment_name ());
```

Comment:
Yeah that sounds good. There are a bunch like this that could be simple static functions.

Reply:
I like the API of having a class do that, but I was unsure of where to put it haha. I'm fine with keeping it as a class especially if it gets more complex when resolving types etc

Reply:
Yeah i would be in favor of grouping them together or turning them into simple functions in another PR